File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.py
from collections import deque
def solution(board, h, w):
    answer = 0
    dh, dw = [0,0,1,-1],[1,-1,0,0]
    for i in range(4):
        nh = h + dh[i]
        nw = w + dw[i]
        if 0<=nh<len(board) and 0<= nw < len(board[0]):
            if board[nh][nw] == board[h][w]:
                answer+=1
    return answer

File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.java
class Solution {
    public int solution(String[][] board, int h, int w) {
        int answer = 0;
        int [] dh ={0,0,-1,1};
        int [] dw ={1,-1,0,0};
        int nw,nh;
        for(int i = 0; i<4; i++){
            nw = w+ dw[i];
            nh = h + dh[i];
            if (0<=nw && nw <board.length && 0<=nh && nh<board[0].length){
                if(board[nh][nw].equals(board[h][w])){
                    answer++;
                }
            }
        }
        return answer;
    }
}